### ISSUE TITLE
[BUG/ISSUE] Cannot find file with valid time stamp in HEMCO

### ISSUE THREAD

**Issue by yantosca** [geoschem/geos-chem#234]
I am submitting this issue on behalf of Shixian Zhai (zhaisx [at] g.harvard.edu):

> I came across a strange problem recently while running GEOS-Chem v12.7.1. If I do simulation for period ahead of 2016 (including 2016), the model runs well. However, if I simulate a period after 2016, then it will have error reminders in the HEMCO.log:
```
HEMCO ERROR: Cannot find field with valid time stamp in 
/n/holylfs/EXTERNAL_REPOS/GEOS-CHEM/gcgrid/data/ExtData/HEMCO/GFED4/v2015-10/2016/GFED4_gen.025x025.201612.nc - 
Cannot get field GFED_TEMP. Please check file name and time (incl. time range flag) in
 ```
My working directory is: 
/n/holyscratch01/jacob_lab/szhai/SUMSANG/geosfp_4x5_tropchem_2019

I can simulate the year 2017 with GEOS-Chem v12.3.1, but now I can not do it with GEOS-Chem v12.7.1.

**Comment by yantosca** [comment]
This seems to be related to the following issues:
- https://github.com/geoschem/geos-chem/issues/233
- https://github.com/geoschem/HEMCO/issues/21

We will work on a fix for this.

**Comment by yantosca** [comment]
After further review, we have discovered that this error is due to the fact that GFED data currently does not exist past 2016.  Also, the GEOS-Chem Steering Committee asked us to modify the HEMCO_Config.rc so that we do not cycle past the end of biomass emissions inventories.

We will shortly be posting beta 2017 and 2018 emissions for use in GEOS-Chem 12.8.0, as described below:

Maggie Marvin (U. Edinburgh) wrote:

> The source GFED beta emissions, which are available only for 2017 and later, are based on the relations between MODIS active fire data and GFED emissions from 2003-2016, so they do a good job of preserving the behavior of the pre-2016 GFED emissions (see attached figs comparing OC emissions between 2014 and 2017). And when we apply our beta emissions processing procedure to the pre-2016 source files, we reproduce the GC-ready emissions as provided in /ExtData/HEMCO/GFED4/v2015-10/ to within 0.00001% for all variables (checked for January 2014).

**Comment by zhendonglu** [comment]
I have exactly the same problem. I don't understand why it need  GFED4_gen.025x025.201612.nc in /GCDATA/ExtData/HEMCO/GFED4/v2015-10/2016/ when I am runing 20180101-20180201. Actually, we have GFED4_gen.025x025.201801.nc in /GCDATA/ExtData/HEMCO/GFED4/v2018-11/2018/

**Comment by lizziel** [comment]
The Harvard ftp site indeed has GFED4 data for 2017 and 2018 that has been used in our group. It is not default in GEOS-Chem and is considered beta. If you would like to use it please reach out to the developers who are listed in the HEMCO/GFED4/v2018-11/README file. They should be able to help determine if it is suitable for your project.

**Comment by zhendonglu** [comment]
> The Harvard ftp site indeed has GFED4 data for 2017 and 2018 that has been used in our group. It is not default in GEOS-Chem and is considered beta. If you would like to use it please reach out to the developers who are listed in the HEMCO/GFED4/v2018-11/README file. They should be able to help determine if it is suitable for your project.

I see, thanks a lot

**Comment by msulprizio** [comment]
In your HEMCO_Config.rc, the default timestamp for the GFED4 data is 1997-2016, which may be why your run is looking for a 2016 file. If you change that to 2018, it should look for the 2018 file.

On a related note, the GFED4.1s beta emissions for 2019-2019 are now available in HEMCO/GFED4/v2020-02 and will be included in 12.8.0 as Bob mentioned above. Those should be used over the files in v2018-11. Please see the [README file](http://ftp.as.harvard.edu/gcgrid/data/ExtData/HEMCO/GFED4/v2020-02/README) for more information. The following changes are required in HEMCO_Config.rc to use those files (note both file path and timestamp change).

```
#==============================================================================
# --- GFED biomass burning emissions (Extension 111)
# NOTE: These are the base emissions in kgDM/m2/s.
#==============================================================================
(((GFED4
111 GFED_TEMP       $ROOT/GFED4/v2020-02/$YYYY/GFED4_gen.025x025.$YYYY$MM.nc  DM_TEMP       1997-2019/1-12/01/0    RF xy kgDM/m2/s * - 1 1
111 GFED_AGRI       $ROOT/GFED4/v2020-02/$YYYY/GFED4_gen.025x025.$YYYY$MM.nc  DM_AGRI       1997-2019/1-12/01/0    RF xy kgDM/m2/s * - 1 1
111 GFED_DEFO       $ROOT/GFED4/v2020-02/$YYYY/GFED4_gen.025x025.$YYYY$MM.nc  DM_DEFO       1997-2019/1-12/01/0    RF xy kgDM/m2/s * - 1 1
111 GFED_BORF       $ROOT/GFED4/v2020-02/$YYYY/GFED4_gen.025x025.$YYYY$MM.nc  DM_BORF       1997-2019/1-12/01/0    RF xy kgDM/m2/s * - 1 1
111 GFED_PEAT       $ROOT/GFED4/v2020-02/$YYYY/GFED4_gen.025x025.$YYYY$MM.nc  DM_PEAT       1997-2019/1-12/01/0    RF xy kgDM/m2/s * - 1 1
111 GFED_SAVA       $ROOT/GFED4/v2020-02/$YYYY/GFED4_gen.025x025.$YYYY$MM.nc  DM_SAVA       1997-2019/1-12/01/0    RF xy kgDM/m2/s * - 1 1

(((GFED_daily
111 GFED_FRAC_DAY   $ROOT/GFED4/v2020-02/$YYYY/GFED4_dailyfrac_gen.025x025.$YYYY$MM.nc GFED_FRACDAY 2003-2019/1-12/1-31/0  RF xy 1 * - 1 1
)))GFED_daily

(((GFED_3hourly
111 GFED_FRAC_3HOUR $ROOT/GFED4/v2020-02/$YYYY/GFED4_3hrfrac_gen.025x025.$YYYY$MM.nc   GFED_FRAC3HR 2003-2019/1-12/1/0-23  RF xy 1 * - 1 1
)))GFED_3hourly
)))GFED4
```

**Comment by zhendonglu** [comment]
> In your HEMCO_Config.rc, the default timestamp for the GFED4 data is 1997-2016, which may be why your run is looking for a 2016 file. If you change that to 2018, it should look for the 2018 file.
> 
> On a related note, the GFED4.1s beta emissions for 2017-2019 are now available on the ftp site in HEMCO/GFED4/v2020-02 and will be included in 12.8.0 as Bob mentioned above (and synced to Compute Canada). Those should be used over the files in v2018-11. Please see the [README file](http://ftp.as.harvard.edu/gcgrid/data/ExtData/HEMCO/GFED4/v2020-02/README) for more information. The following changes are required in HEMCO_Config.rc to use those files (note both file path and timestamp change).
> 
> ```
> #==============================================================================
> # --- GFED biomass burning emissions (Extension 111)
> # NOTE: These are the base emissions in kgDM/m2/s.
> #==============================================================================
> (((GFED4
> 111 GFED_TEMP       $ROOT/GFED4/v2020-02/$YYYY/GFED4_gen.025x025.$YYYY$MM.nc  DM_TEMP       1997-2019/1-12/01/0    RF xy kgDM/m2/s * - 1 1
> 111 GFED_AGRI       $ROOT/GFED4/v2020-02/$YYYY/GFED4_gen.025x025.$YYYY$MM.nc  DM_AGRI       1997-2019/1-12/01/0    RF xy kgDM/m2/s * - 1 1
> 111 GFED_DEFO       $ROOT/GFED4/v2020-02/$YYYY/GFED4_gen.025x025.$YYYY$MM.nc  DM_DEFO       1997-2019/1-12/01/0    RF xy kgDM/m2/s * - 1 1
> 111 GFED_BORF       $ROOT/GFED4/v2020-02/$YYYY/GFED4_gen.025x025.$YYYY$MM.nc  DM_BORF       1997-2019/1-12/01/0    RF xy kgDM/m2/s * - 1 1
> 111 GFED_PEAT       $ROOT/GFED4/v2020-02/$YYYY/GFED4_gen.025x025.$YYYY$MM.nc  DM_PEAT       1997-2019/1-12/01/0    RF xy kgDM/m2/s * - 1 1
> 111 GFED_SAVA       $ROOT/GFED4/v2020-02/$YYYY/GFED4_gen.025x025.$YYYY$MM.nc  DM_SAVA       1997-2019/1-12/01/0    RF xy kgDM/m2/s * - 1 1
> 
> (((GFED_daily
> 111 GFED_FRAC_DAY   $ROOT/GFED4/v2020-02/$YYYY/GFED4_dailyfrac_gen.025x025.$YYYY$MM.nc GFED_FRACDAY 2003-2019/1-12/1-31/0  RF xy 1 * - 1 1
> )))GFED_daily
> 
> (((GFED_3hourly
> 111 GFED_FRAC_3HOUR $ROOT/GFED4/v2020-02/$YYYY/GFED4_3hrfrac_gen.025x025.$YYYY$MM.nc   GFED_FRAC3HR 2003-2019/1-12/1/0-23  RF xy 1 * - 1 1
> )))GFED_3hourly
> )))GFED4
> ```

Thank you. It helps a lot.
